tests/io/test_data_catalog.py

```diff
@@ -309,12 +309,13 @@ def test_mutating_datasets_not_allowed(self, data_catalog_from_config):
 
     def test_confirm(self, mocker, caplog):
         """Confirm the dataset"""
+        caplog.set_level(logging.INFO)
         mock_ds = mocker.Mock()
         data_catalog = DataCatalog(data_sets={"mocked": mock_ds})
         data_catalog.confirm("mocked")
         mock_ds.confirm.assert_called_once_with()
         assert caplog.record_tuples == [
-            ("kedro.io.data_catalog", logging.INFO, "Confirming DataSet 'mocked'")
+            ("kedro.io.data_catalog", logging.INFO, "Confirming DataSet [dark_orange]mocked[/dark_orange]")
```

Comment:
I don't think this is the best way to modify this test. When run, the captured stdout call reads without the rich style tags, but they are still present in the captured log call:

> tests\io\test_data_catalog.py:316: AssertionError (this error is from before the most recent commit)
---------------------------- Captured stdout call -----------------------------
[07/02/22 19:17:52] INFO     Confirming DataSet mocked      data_catalog.py:600
------------------------------ Captured log call ------------------------------
INFO     kedro.io.data_catalog:data_catalog.py:600 Confirming DataSet [dark_orange]mocked[/dark_orange]

Ideally the test should be modified to compare the captured stdout call as opposed to the captured log call to avoid specifying the style tags in the test. Potentially [capsys](https://doc.pytest.org/en/latest/how-to/capture-stdout-stderr.html#accessing-captured-output-from-a-test-function) could be used to do. The following replacement was attempted but would only yield an empty string:

```
def test_confirm(self, mocker, capsys):
        """Confirm the dataset"""
        mock_ds = mocker.Mock()
        data_catalog = DataCatalog(data_sets={"mocked": mock_ds})
        data_catalog.confirm("mocked")
        mock_ds.confirm.assert_called_once_with()
        assert capsys.readouterr().out == [
            ("kedro.io.data_catalog", logging.INFO, "Confirming DataSet 'mocked'")
        ]
```
The test would fail with the following output:
> E       assert '' == [('kedro.io.d...et 'mocked'")]
E         +''
E         -[('kedro.io.data_catalog', 20, "Confirming DataSet 'mocked'")]

Other ideas/approaches are very welcome 😅

Reply:
One quick question, does the test result depends on the terminal type (Dumb vs Emulated Terminal?)

Reply:
I actually think is exactly the right way to modify the test. We need to think about what we're actually testing for here: this is a unit test to ensure the correct behaviour of the `confirm` method. This is covered by the `assert_called_once_with`, and the benefit of testing logging at all after that is debatable. But let's say we want to check that `confirm` also outputs the correct log message. Then what we're interested in is "are we making the right call to `logging.INFO`", not "how does rich render that call".

Hence by including the `dark_orange` styling in the test, you're testing (a bit like a mock) whether the call to `logging.INFO` is correct. We can trust rich will figure out to strip that styling in dumb terminals (as you tested manually), as that is a separate, well-tested, third party library. If we don't include `dark_orange` in the test then you would not be testing the actual call we're making. Put another way: the fact that this passes on CI and your local machine is actually a good demonstration that you're testing the right thing here, i.e. the logging call itself and not how it gets rendered.

I very much recommend[ this series of articles on TDD in Python](https://www.thedigitalcatonline.com/blog/2020/09/10/tdd-in-python-with-pytest-part-1/). We don't really follow it strictly on kedro, but it's definitely worth understanding. The case here would be one of an [outgoing command](https://www.thedigitalcatonline.com/blog/2020/09/15/tdd-in-python-with-pytest-part-3/#outgoing-queries-and-commands).

Reply:
![image](https://user-images.githubusercontent.com/18221871/177136185-62192ab3-7016-4f16-94d5-27886785f275.png)

I believe these styling tags should only have an effect on `stdout`, and shouldn't be logged in files.

Reply:
> One quick question, does the test result depends on the terminal type (Dumb vs Emulated Terminal?)

Nope, it passes on both